webhook/resources/objectstore/mutator.go

```diff
@@ -0,0 +1,85 @@
+package objectstore
+
+import (
+	"fmt"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	"github.com/longhorn/longhorn-manager/webhook/common"
+	"github.com/pkg/errors"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type objectStoreMutator struct {
+	admission.DefaultMutator
+	ds *datastore.DataStore
+}
+
+func NewMutator(ds *datastore.DataStore) admission.Mutator {
+	return &objectStoreMutator{ds: ds}
+}
+
+func (osm *objectStoreMutator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "objectstores",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.ObjectStore{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (osm *objectStoreMutator) Create(req *admission.Request, obj runtime.Object) (ops admission.PatchOps, err error) {
+	store, ops, err := getDefaultMutations(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if store.Spec.TargetState == "" {
+		ops = append(ops, "{\"op\": \"add\", \"path\": \"/spec/targetState\", \"value\": \"running\"}")
+	}
+
+	if store.Spec.Image == "" {
+		imageSetting, err := osm.ds.GetSetting("object-store-image")
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read setting \"object-store-image\"")
+		}
+		ops = append(ops, fmt.Sprintf("{\"op\": \"add\", \"path\": \"/spec/image\", \"value\": \"%v\"}", imageSetting.Value))
+	}
+
+	if store.Spec.UiImage == "" {
+		uiImageSetting, err := osm.ds.GetSetting("object-store-ui-image")
```

Comment:
NIT: `GetSettingWithAutoFillingRO`

Reply:
NIT
```suggestion
		uiImageSetting, err := osm.ds.GetSetting(types.SettingNameObjectStoreUIImage)
```

Reply:
Please explain, a code search yielded no results for this